packages/core/src/mixins/closable.js

```diff
@@ -108,40 +92,44 @@ const closable = ({
             }
         }
 
-        handleDocumentClickOrTouch = (event) => {
+        /**
+         * @param {React.MouseEvent} event
+         */
+        handleOuterLayerClick = (event) => {
             const options = this.getOptions();
 
-            if (this.clickedInside) {
-                // already scheduled close when clicked inside, skip here.
-                this.clickedInside = false;
-                return;
-            }
-
             if (options.onClickOutside) {
-                this.delayedClose(event);
+                this.props.onClose(event);
             }
         }
 
-        handleInsideClickOrTouch = (event) => {
+        /**
+         * @param {React.MouseEvent} event
+         */
+        handleInsideClick = (event) => {
             const options = this.getOptions();
-            this.clickedInside = true;
+            event.stopPropagation();
```

Comment:
This stops event from propagating at the `closable()` layer.
I was thinking maybe it's not a good idea because it's hard to realize that you're event is blocked here.